gmso/core/topology.py

```diff
@@ -960,13 +1154,86 @@ def get_index(self, member):
+        parameter : a string
+            A string determining what aspects of the gmso topology will be reported. Options are: 'sites','bonds','angles',
+        'dihedrals', and 'impropers'. Defaults to 'sites'.
+        site_attrs : List of strings that are attributes of the topology site and can be included as entries in the pandas dataframe.
+            Examples of these can be found by printing `topology.sites[0].__dict__`.
+            See https://gmso.mosdef.org/en/stable/data_structures.html#gmso.Atom for additional information on labeling.
+        unyts_bool: bool, default=True
+            Determine if numerical values are saved as unyt quantities or floats. See
+        https://unyt.readthedocs.io/en/stable/usage.html
+        for more information about manipulating unyt quantities.
+        Default is True.
+
+        Returns
+        -------
+        Pandas Dataframe
+            A pandas.Dataframe object, see https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html
+            for further information.
+
+        Examples
+        ________
+        >>> topology.to_dataframe(parameter = 'sites', site_attrs = ['charge'])
+            This will return a dataframe with a listing of the sites and include the charges that correspond to each site.
+        >>> topology.to_dataframe(parameter = 'dihedrals', site_attrs = ['positions'])
+            This will return a dataframe with a listing of the sites that make up each dihedral, the positions of each of
+            those sites, and the parameters that are associated with the dihedrals.
+
+        Notes
+        ____
+        A dataframe is easily manipulated. In order to change the rounding to two decimals places for a column named `label`:
+            >>> df['label'] = df['label'].round(2)
+        The column labels can also be easily modified. This line can take a dataframe `df` and rename a column labeled
+        `Atom0` to `newname` using a dictionary.
+            >>> df.rename(columns = {'Atom0':'newname'})
+        See https://pandas.pydata.org/pandas-docs/stable/getting_started/intro_tutorials/index.html for further information.
+        """
+        from gmso.utils.io import import_
+
+        pd = import_("pandas")
+        if not site_attrs:
+            site_attrs = []
+        df = pd.DataFrame()
+        if not self.is_typed:
+            raise GMSOError(
+                "This topology is not typed, please type this object before converting to a pandas dataframe"
+            )
+        if parameter == "sites":
+            df["atom_types"] = list(site.atom_type.name for site in self.sites)
+            df["names"] = list(site.name for site in self.sites)
+            for attr in site_attrs:
+                df = self._parse_dataframe_attrs(
+                    df, attr, parameter, unyts_bool
+                )
+        elif parameter in ["bonds", "angles", "dihedrals", "impropers"]:
+            df = self._pandas_from_parameters(
```

Comment:
Probably good to have a check here that is like:

`if len(self.bonds) !=0`

for each of the possible parameters so that we can give a descriptive warning as to why there will be nothing instead of just erroring out.

Reply:
Addressed in d21d2d6b771209a4d2e1b4770267271707f89b17